src/components/userGroups/K8sObjectPermissions/K8sPermissionModal.tsx

```diff
@@ -0,0 +1,153 @@
+                k8PermissionList[key].action = data
+                break
+            default:
+                break
+        }
+
+        setPermissionList([...k8PermissionList])
+    }
+
+    const stopPropogation = (e) => {
+        e.stopPropagation()
+    }
+
+    const addNewPermissionCard = () => {
+        handleK8sPermission('add')
+    }
+
+    const savePermission = () => {
+        setK8sPermission((prev) => {
+            if (selectedPermissionAction?.action === 'edit') {
+                prev[selectedPermissionAction.index] = k8PermissionList[0]
+                return [...prev]
+            }
+            return [...prev, ...k8PermissionList]
+        })
+        close(false)
+    }
+
+    return (
+        <VisibleModal className="" close={close}>
+            <div onClick={stopPropogation} className="modal-body--ci-material h-100 dc__overflow-hidden">
+                <div className="flex pt-12 pb-12 pl-20 pr-20 dc__content-space dc__border-bottom">
+                    <span className="flex left fw-6 lh-24 fs-16">Kubernetes object permission</span>
+                    <span className="icon-dim-20 cursor" onClick={close}>
+                        <Close />
+                    </span>
+                </div>
+                <div className="p-20 fs-13 dc__overflow-scroll dc__cluster-modal">
+                    {!selectedPermissionAction && (
+                        <div className="anchor pointer flex left fs-13 fw-6" onClick={addNewPermissionCard}>
+                            <AddIcon className="add-svg mr-12" /> Add another
+                        </div>
+                    )}
+                    {k8PermissionList?.map((_k8sPermission, index) => {
+                        return (
+                            <K8sListItemCard
+                                k8sPermission={_k8sPermission}
+                                handleK8sPermission={handleK8sPermission}
+                                index={index}
+                                namespaceMapping={namespaceMapping}
+                                setNamespaceMapping={setNamespaceMapping}
+                                apiGroupMapping={apiGroupMapping}
+                                setApiGroupMapping={setApiGroupMapping}
+                                kindMapping={kindMapping}
+                                setKindMapping={setKindMapping}
+                                objectMapping={objectMapping}
+                                setObjectMapping={setObjectMapping}
+                            />
+                        )
+                    })}
+                </div>
+                <div className="w-100 pt-16 pb-16 pl-20 pr-20 flex right dc__border-top">
+                    <button type="button" className="cta cancel h-36 flex mr-16" disabled={false} onClick={close}>
+                        Cancel
+                    </button>
+                    <ButtonWithLoader
+                        rootClassName="cta cta--workflow"
```

Comment:
I think we'll not get the class `cta--workflow` properties unless it's in some global css

Reply:
done